src/storage/core.rs

```diff
@@ -712,13 +729,39 @@ where
                             .with_context(|| {
                                 anyhow!(format!("failed to save corrupted blob {:?}", file))
                             })?;
+                        corrupted += 1;
                     } else {
                         return Err(e.context(msg));
                     }
                 }
             }
         }
-        Ok(blobs)
+        Ok((blobs, corrupted))
+    }
+
+    async fn count_old_corrupted_blobs(files: &[DirEntry], config: &Config) -> usize {
```

Comment:
I think it is better to take the path to the pearl directory rather than all of its content, because we only need one directory inside of it. You can combine pearl directory with `config.corrupted_dir_name()` and get an expected path. Then you can easily create `Path` from it, check its exists and then call `read_dir`